cmd/mdatagen/templates/telemetry.go.tmpl

```diff
@@ -133,4 +133,14 @@ func NewTelemetryBuilder(settings component.TelemetrySettings, options ...teleme
     return &builder, errs
 }
 
+{{- range $name, $metric := .Telemetry.Metrics }}
+{{- if not $metric.Data.Async }}
+
+func (b *TelemetryBuilder) Record{{ $name.Render }}(ctx context.Context, val {{ $metric.Data.BasicType }}, opts ...metric.{{- if eq $metric.Data.Type "Histogram" -}}Record{{- else -}}Add{{- end -}}Option) {
+    b.{{ $name.Render }}.{{- if eq $metric.Data.Type "Histogram" -}}Record{{- else -}}Add{{- end -}}(ctx, val, opts...)
```

Comment:
I believe we also need `Record` for gauge metrics.

I found that the [Int64Gauge](https://pkg.go.dev/go.opentelemetry.io/otel/metric#Int64Gauge) metric type has `Record`, not `Add` as rendered here. (Same goes for Float64Gauge) This is hit if you do:

```
telemetry:
  metrics:
    fake_metric:
      enabled: true
      description: Description of metric
      unit: "{x}"
      gauge:
        value_type: int
```